evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java

```diff
@@ -170,7 +206,7 @@ static int readBigEndianU16(final int index, final byte[] array) {
     if (index + 1 >= array.length) {
       throw new IndexOutOfBoundsException();
     }
-    return ((array[index] & 0xff) << 8) | (array[index + 1] & 0xff);
+    return ((array[index] << 8) & 0xff00) | (array[index + 1] & 0xff);
```

Comment:
This seems like the same result as before but masks the full 16 bits before or'ing the lower half. Whats the preference behind the change?

Reply:
From different drafts of the EOF fork I was working on.  This fromulation has less issues with sign extension when you do the masking last, and makes the byte position clear. More of an issue with right shifts but masking last has reduced errors in other code I've worked on.